src/Time/TakeStep.hpp

```diff
@@ -0,0 +1,26 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Time/Actions/ChangeStepSize.hpp"
+#include "Time/Actions/RecordTimeStepperData.hpp"
+#include "Time/Actions/UpdateU.hpp"
+#include "Utilities/Gsl.hpp"
+
+/// Bundled method for recording the current system state in the history, and
+/// updating the evolved variables and step size.
+///
+/// This function is used to encapsulate any needed logic for updating the
+/// system, and in the case for which step parameters may need to be rejected
```

Comment:
I don't understand where the step unwinding is happening. I was expecting a loop inside the `take_step` function surrounding the `update_u` and `change_step_size` calls, but that appears to be absent. What am I missing?

Reply:
Ah, yes, good catch -- I hadn't added the loop yet because in the initial version, none of the steppers were able to reject a step, but now CFL can, so this should be updated.